### DISCUSSION TITLE
Is fetching top tracks from lastfm still supported?

### DISCUSSION THREAD

**Discussion by xusiai** [navidrome/navidrome#3540] [Q&A]
For the past few months Navidrome has not shown top tracks for artists. Fetching artist information has also been hit and miss (seem to work for some whilst others show nothing but a lastfm link where information does exist). I rotated lastfm credentials but this did not resolve it. I had a look around the discussion and issue boards but couldn't find an answer. Is this a known issue or perhaps a new limitation on lastfm's API?

**Comment by kgarner7** [comment]
The lastfm link _always_ exists (unless lastfm is disabled entirely), but may not be accurate (it is just the artist name). Most likely, the artists you are searching aren't being found (or not with sufficient metadata).

**Reply by xusiai** [reply]
I do not think it is case of artist matching failing or data missing on lastfm since top tracks have stopped loading for _all artists_ - even the ones who import _about_ information correctly.

**Reply by kgarner7** [reply]
It definitely still works (I just tested it myself). The lastfm endpoint (for a popular artist in my library) works, and then tracks from that are resolved by
1. MusicBrainz id (if it exists, very rarely)
2. Exactly matching sort name.

Your mileage may vary if you have tracks that match this.

**Comment by xusiai** [comment]
Example of _about_ information being shown but no top tracks (which do exist on the lastfm artist page). 
![2024-12-14T15:16:47,863996115+01:00](https://github.com/user-attachments/assets/7859bcc6-da8f-47eb-8c6b-9e8062795a28)

**Reply by metalheim** [reply]
The navidrome webui never had topSongs implemented. So this did not stop working, it never did.

Other clients (Android Apps using subsonic API, Desktop Apps like feishin) have it implemented and it still works

**Reply by xusiai** [reply]
I use Feishin but assumed this was Navidrome related. This problem still occur for me; top songs do not work in Feishin, but since this is not Navidrome, I will head over to the Feishin repo.
